### ISSUE TITLE
Use overflow checked arithmetic in effects of the CT

### ISSUE THREAD

**Issue by LuisPH3** [Fantom-foundation/Tosca#493]
There are operations in the effects of the specification that do not check for overflows.
mostly related to dynamicGas computation, but may be the case that there are more.

**Comment by facuMH** [comment]
After #613 is merged, I think this issue should be closed, since that PR handles most dynamic gas overflow checks, and after a revision of the effects, I believe (could be wrong), that no other overflows could be triggered.

**Comment by LuisPH3** [comment]
As discussed, #613 handles this issue. If further errors are found, we can make a new issue.
